bmc/firmware.go

```diff
@@ -2,201 +2,163 @@ package bmc
 
 import (
 	"context"
-	"errors"
 	"fmt"
 	"io"
 
-	"github.com/hashicorp/go-multierror"
-)
+	bmclibErrs "github.com/bmc-toolbox/bmclib/errors"
 
-// BMCVersionGetter retrieves the current BMC firmware version information
-type BMCVersionGetter interface {
-	GetBMCVersion(ctx context.Context) (version string, err error)
-}
+	"github.com/pkg/errors"
 
-// BMCFirmwareUpdater upgrades the BMC firmware
-type BMCFirmwareUpdater interface {
-	FirmwareUpdateBMC(ctx context.Context, fileReader io.Reader, fileSize int64) (err error)
-}
+	"github.com/hashicorp/go-multierror"
+)
 
-// BIOSVersionGetter retrieves the current BIOS firmware version information
-type BIOSVersionGetter interface {
-	GetBIOSVersion(ctx context.Context) (version string, err error)
+// FirmwareInstaller defines an interface to install firmware updates
+type FirmwareInstaller interface {
+	// FirmwareInstall uploads firmware update payload to the BMC returning the task ID
+	FirmwareInstall(ctx context.Context, component, applyAt string, forceInstall bool, reader io.Reader) (jobID string, err error)
```

Comment:
nit: the below `FirmwareInstall()` func returns a `taskID`, should this interface match that semantic?